packages/core-mobile/app/services/swap/SwapService.ts

```diff
@@ -73,49 +95,65 @@ class SwapService {
     network,
     account,
     sentryTrx
-  }: SwapRate): Promise<OptimalRate | APIError> {
+  }: SwapRate): Promise<OptimalRate | Error> {
     return SentryWrapper.createSpanFor(sentryTrx)
       .setContext('svc.swap.get_rate')
       .executeAsync(async () => {
         if (network.isTestnet) {
-          throw NETWORK_UNSUPPORTED_ERROR
+          throw TESTNET_NETWORK_UNSUPPORTED_ERROR
+        }
+        if (!SUPPORTED_SWAP_NETWORKS.includes(network.chainId)) {
+          throw new Error(`${network.chainName} is not supported by Paraswap`)
         }
         if (!account) {
           throw new Error('Account address missing')
         }
 
-        const optimalRates = async (): Promise<OptimalRate | APIError> => {
-          return await this.paraSwap.getRate(
+        const optimalRates = async (): Promise<OptimalRate | Error> => {
+          return await this.getParaSwapSDK(network.chainId).swap.getRate({
             srcToken,
             destToken,
-            srcAmount,
-            account.addressC,
-            swapSide,
-            {},
+            amount: srcAmount,
+            userAddress: account.addressC,
+            side: swapSide,
             srcDecimals,
             destDecimals
-          )
+          })
         }
```

Comment:
I compared the `getRate` functions between the old and new sdk, and it seems not to return `Error` anymore. same for the `getSpender()` in the function below.

Reply:
updated [here](3f8a29109e9e02dfb3c3e6ad6e4210d4fe81c60c)